File: golang/27.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(removeElement([]int{3, 2, 2, 3}, 3))
	fmt.Println(removeElement([]int{0, 1, 2, 2, 3, 0, 4, 2}, 2))
}

func removeElement(nums []int, val int) int {
	i := 0
	for idx, num := range nums {
		if num != val {
			nums[i] = nums[idx]
			i++
		}
	}
	fmt.Println(nums, val)
	return i
}

File: python/27.py
from typing import List


def removeElement(nums: List[int], val: int) -> int:
    count = 0
    for i in range(len(nums)):
        if(nums[i] != val):
            nums[count] = nums[i]
            count += 1
    print(nums)
    return count 

print(removeElement([3,2,2,3], 3))
print(removeElement([0,1,2,2,3,0,4,2], 2))